arborist/policy.go

```diff
@@ -179,25 +176,16 @@ func (policy *Policy) validate() *ErrorResponse {
 	return nil
 }
 
-// createInDb writes out the policy to the database.
-func (policy *Policy) createInDb(tx *sqlx.Tx) *ErrorResponse {
-	errResponse := policy.validate()
-	if errResponse != nil {
-		return errResponse
-	}
+// add_resources_and_roles takes a policy and links it in the database
+// to each of its resources and roles.
+func (policy *Policy) add_resources_and_roles(tx *sqlx.Tx) *ErrorResponse {
 
 	var policyID int
-	// TODO: make sure description works as expected
-	stmt := "INSERT INTO policy(name, description) VALUES ($1, $2) RETURNING id"
-	row := tx.QueryRowx(stmt, policy.Name, policy.Description)
-	err := row.Scan(&policyID)
+	stmt := "SELECT id FROM policy WHERE name = $1"
```

Comment:
we could pass in the policy ID instead of selecting here

Reply:
Ah yes. I was thinking: (1) It's a bit weird to require an ID argument when the helper fn technically doesn't need it (gets name from policy and does SELECT), and (2) if any other fn calls this in future and doesn't already have the ID it will have to then look up the ID. But I guess this is all internal code so (2) doesn't really matter. Will change!

Reply:
Thanks! Further notes: Though it's hard to balance, (2) is usually the way to over-designing.